src/main/java/net/neoforged/neoforge/network/registration/NetworkRegistry.java

```diff
@@ -79,6 +86,7 @@ public class NetworkRegistry {
     private static final Logger LOGGER = LogUtils.getLogger();
 
     private static final AttributeKey<NetworkPayloadSetup> ATTRIBUTE_PAYLOAD_SETUP = AttributeKey.valueOf("neoforge:payload_setup");
+    private static final AttributeKey<Set<ResourceLocation>> ATTRIBUTE_ADHOC_CHANNELS = AttributeKey.valueOf("neoforge:adhoc_channels");
     private static final AttributeKey<Boolean> ATTRIBUTE_IS_MODDED_CONNECTION = AttributeKey.valueOf("neoforge:is_modded_connection");
```

Comment:
This should be replaced by a ConnectionType. Then we could also directly get the ConnectionType from the channel, and not need a second parameter for the packet filters.